### PR TITLE
ci: Cleanup and factorize a few functions in the integration tests

### PR DESCRIPTION
**Pull request by sboeuf**
This PR simplifies a bit the integration tests code by replacing `ssh_command_ok()` with `ssh_command()` now that it returns an actual error if the exit code is not 0.
Also, it factorizes some code executing commands on the host, by introducing two new wrapper functions in `test_infra`.

### CODE REVIEW

File: test_infra/src/lib.rs [hunk 1]
```diff
@@ -589,3 +590,17 @@ pub fn ssh_command_ip(
...
+        .args(&["-c", command])
+        .status()
+        .expect(&format!("Expected '{}' to work", command))
+}
+
+pub fn exec_host_command_output(command: &str) -> Output {
+    std::process::Command::new("bash")
+        .args(&["-c", command])
+        .output()
+        .expect(&format!("Expected '{}' to work", command))
```
Comment (rbradford) [inline test_infra/src/lib.rs:605]: You don't actually check the exit value here.

File: test_infra/src/lib.rs [hunk 2]
```diff
@@ -589,3 +590,17 @@ pub fn ssh_command_ip(
         timeout,
     )
 }
+
+pub fn exec_host_command_status(command: &str) -> ExitStatus {
+    std::process::Command::new("bash")
+        .args(&["-c", command])
+        .status()
+        .expect(&format!("Expected '{}' to work", command))
```
Comment (rbradford) [inline test_infra/src/lib.rs:598]: This is only the error from the exec and wait. So the error message is a bit confusing here.

**Review by rbradford — COMMENTED** [review]
I think from a usability perspective it's better to check the exit codes in those helper functions. The user probably wants those to return non-zero.

**Comment by rbradford** [comment]
> I think from a usability perspective it's better to check the exit codes in those helper functions. The user probably wants those to return non-zero.

The assert on the `.success()` looks kinda okay so this is probably fine as is

File: tests/integration.rs [hunk 3]
```diff
@@ -6176,7 +6035,7 @@ mod tests {
                 guest.wait_vm_boot(None).unwrap();
 
                 // Check if SGX is correctly detected in the guest.
-                assert!(guest.check_sgx_support().unwrap());
+                assert!(guest.check_sgx_support().is_ok());
```
Comment (rbradford) [inline tests/integration.rs:6038]: Asserting on a boolean like this loses the underlying error. Using `.expect()` or just `.unwrap()` would make the underlying error visible in the logs.

Comment (sboeuf) [reply tests/integration.rs:6038]: Ah yes I understand that's indeed better. Let me unwrap all these.

File: test_infra/src/lib.rs (hunk 2 above)
Comment (sboeuf) [reply test_infra/src/lib.rs:598]: Yes it should be more something like `Expected '{}' to run`.

File: test_infra/src/lib.rs (hunk 1 above)
Comment (sboeuf) [reply test_infra/src/lib.rs:605]: Same here, I'll change the message.

**Review by rbradford — APPROVED** [review]

**Pull request merged**